### PR TITLE
Min & Max key functions

### PR DESCRIPTION
**Pull request by sumerman**
Adds min_key & max_key funcitons to istore:

```SQL
    SELECT min_key('1=>1, 2=>1'::istore);
     min_key
    ---------
           1
    (1 row)

    SELECT max_key('1=>1, 2=>1'::istore);
     max_key
    ---------
           2
    (1 row)

```

### CODE REVIEW

File: spec/istore/migration_spec.rb [hunk 1]
```diff
@@ -1,18 +1,20 @@
 require 'spec_helper'
-describe "isagg" do
 
+describe "istore migrations" do
   it 'should be upgradeable' do
-    query("CREATE EXTENSION istore VERSION '0.1.3'")
-    query("ALTER EXTENSION istore UPDATE TO '0.1.4'")
-    query("ALTER EXTENSION istore UPDATE TO '0.1.3'")
```
Comment (rapimo) [inline spec/istore/migration_spec.rb]: I'd leave theses upgrade function to still test old upgrade paths

File: spec/istore/migration_spec.rb [hunk 2]
```diff
@@ -1,18 +1,20 @@
...
-    "istore", "0.1.3"
+    "istore", "0.1.6"
+    query("ALTER EXTENSION istore UPDATE TO '0.1.5'")
+    query("SELECT e.extname, e.extversion FROM pg_catalog.pg_extension e WHERE e.extname = 'istore'ORDER BY 1;").should match\
+    "istore", "0.1.5"
   end
 
   it 'should be downgradeable' do
     query "CREATE EXTENSION istore;"
-    query "ALTER EXTENSION istore UPDATE TO '0.1.3';"
```
Comment (rapimo) [inline spec/istore/migration_spec.rb]: if you just leave that you implicitly test 0.1.6 -> 0.1.5 as well

File: sql/bigistore.sql [hunk 3]
```diff
@@ -17,6 +17,16 @@ RETURNS SETOF record
 AS 'istore','bigistore_each'
 LANGUAGE C STRICT IMMUTABLE;
 
+CREATE FUNCTION smallest_key(bigistore)
```
Comment (rapimo) [inline sql/bigistore.sql]: how about naming these `first_key` and `last_key` ?

File: src/bigistore.c [hunk 4]
```diff
@@ -98,6 +98,46 @@ bigistore_apply_datum(BigIStore *arg1, Datum arg2, PGFunction applyfunc)
...
+ */
+PG_FUNCTION_INFO_V1(bigistore_smallest_key);
+Datum
+bigistore_smallest_key(PG_FUNCTION_ARGS)
+{
+    BigIStore *istore;
+	int32   key;
+
+    istore = PG_GETARG_BIGIS(0);
+	if (istore->len == 0) {
```
Comment (rapimo) [inline src/bigistore.c]: surrounding code style define brace on new line

Comment (rapimo) [reply src/bigistore.c]: also indent using spaces

File: .gitignore [hunk 5]
```diff
@@ -23,4 +23,5 @@
 *.swp
 temp
 results
-bench/benchmarks*
\ No newline at end of file
+bench/benchmarks*
+vendor
```
Comment (rapimo) [inline .gitignore]: what's the intention of this folder?

File: src/depcode.c [hunk 6]
```diff
@@ -1,6 +1,7 @@
 #include "istore.h"
 #include "funcapi.h"
 #include "utils/array.h"
+#include "utils/builtins.h"
```
Comment (rapimo) [inline src/depcode.c]: that's already included in `istore.h`

File: src/istore_cast.c [hunk 7]
```diff
@@ -3,7 +3,7 @@
  */
 
 #include "istore.h"
-#include "istore.h"
+#include "utils/builtins.h"
```
Comment (rapimo) [inline src/istore_cast.c]: see above

File: src/pairs.c [hunk 8]
```diff
@@ -1,5 +1,6 @@
 #include "istore.h"
 #include "avl.h"
+#include "utils/builtins.h"
```
Comment (rapimo) [inline src/pairs.c]: same

File: .gitignore (hunk 5 above)
Comment (sumerman) [reply .gitignore]: I prefer `bundle install --path vendor/bundle` to global installs. One more path in `.gitignore` doesn't hurt, does it?

File: spec/istore/migration_spec.rb (hunk 1 above)
Comment (sumerman) [reply spec/istore/migration_spec.rb]: For some reason, I thought it' supposed to test the most recent path. Will revert this.

File: sql/bigistore.sql (hunk 3 above)
Comment (sumerman) [reply sql/bigistore.sql]: It's how I initially named those functions but then changed the naming as I realized that `first`/`last` involves reasoning about `istore` ordering (it's always sorted, but one have to keep that in mind) while `smallest`/`biggest` is very explicit.

File: src/depcode.c (hunk 6 above)
Comment (sumerman) [reply src/depcode.c]: Leftovers from before the master was made 10.0 compatible. Will remove.

**Comment by sumerman** [comment]
Rebased/squashed everything into one commit that addresses everything except naming and gitignore.

File: .gitignore (hunk 5 above)
Comment (rapimo) [reply .gitignore]: nope was just curious

File: sql/bigistore.sql (hunk 3 above)
Comment (rapimo) [reply sql/bigistore.sql]: mmh  `min_key`  `max_key` ?

Comment (sumerman) [reply sql/bigistore.sql]: i like it

**Comment by sumerman** [comment]
Renamed and rebased.

**Pull request merged**